src/test/java/net/kaczmarzyk/spring/data/jpa/web/utils/RequestAttributesWithPathVariablesUtil.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2014-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.web.utils;
+
+import net.kaczmarzyk.spring.data.jpa.web.MockWebRequest;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.servlet.HandlerMapping;
+
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Util for setting path variables in request's attributes
+ * It's used in tests that should work with new request path variable resolving approach in {@link net.kaczmarzyk.spring.data.jpa.web.WebRequestProcessingContext WebRequestProcessingContext}
```

Comment:
"new" will become almost meaningless once we merge this PR. Please add a description (here and/or in the WebRequestProcessingContext) that will actually explain (I still don't get it, after third reading of this PR):
1. Why the standard approach needs fallback (and when it can happen)
2. how fallback works and why it can fail (with example).